src/convert_xml_to_csv.py

```diff
@@ -0,0 +1,80 @@
+import csv
+import sys
+import xml.etree.ElementTree as ET
+from datetime import datetime
+from typing import List
+
+
+def check_leap_year(year: int):
+    if year % 400 == 0 or year % 100 != 0 and year % 4 == 0:
+        return True
+    else:
+        return False
+
+
+def check_hired_greater_than_3year(hired_str: str):
+    hired = datetime.strptime(hired_str, "%Y-%m-%d")
+    current = datetime.now()
+    diff_year = current.year - hired.year
+    if diff_year >= 3:
+        return True
+
+    days_per_month = 366 / 12 if check_leap_year(hired.year) else 365 / 12
+    current_days = (days_per_month * current.month) + current.day
+    hired_days = (days_per_month * hired.month) + hired.day
+    if current_days - hired_days == 0:
+        return True
+
+    return False
+
+
+def read_xml_to_list_of_dict(path: str) -> tuple[List[dict], List[dict]]:
+    tree = ET.parse(path)
+    root = tree.getroot()
+    dev_mountain = []
+    dev_club = []
+    for child in root.iter('record'):
+        row = {}
+        for elem in child.iter():
+            if elem.tag == "record":
+                continue
+
+            if elem.tag == "BIRTHDAY" or elem.tag == "HIRED":
+                row[elem.tag] = datetime.strptime(elem.text, "%d-%m-%Y").strftime("%Y-%m-%d")
+                continue
+
+            row[elem.tag] = elem.text
+        if check_hired_greater_than_3year(row["HIRED"]) \
+                and row["POSITION"] in ["Airhostess", "Pilot", "Steward"] \
+                and row["STATUS"] == "1" \
+                and row["GENDER"] in ["0", "1"]:
+            dev_club.append(row)
+            continue
+
+        if row["STATUS"] not in ["1", "2", "3"] and row["GENDER"] not in ["0", "1"]:
+            continue
+
+        dev_mountain.append(row)
+    return dev_mountain, dev_club
```

Comment:
following Clean code book by Uncle Bob. you should create new variables to keep them be searchable
```suggestion
def read_xml_to_list_of_dict(path: str) -> tuple[List[dict], List[dict]]:
    STATUS = ["1", "2", "3"]
    GENDER = ["0", "1"]
    POSITION = ["Airhostess", "Pilot", "Steward"]
    tree = ET.parse(path)
    root = tree.getroot()
    dev_mountain = []
    dev_club = []
    for child in root.iter('record'):
        row = {}
        for elem in child.iter():
            if elem.tag == "record":
                continue

            if elem.tag == "BIRTHDAY" or elem.tag == "HIRED":
                row[elem.tag] = datetime.strptime(elem.text, "%d-%m-%Y").strftime("%Y-%m-%d")
                continue

            row[elem.tag] = elem.text
        if check_hired_greater_than_3year(row["HIRED"]) \
                and row["POSITION"] in POSITION \
                and row["STATUS"] == "1" \
                and row["GENDER"] in GENDER:
            dev_club.append(row)
            continue

        if row["STATUS"] not in STATUS and row["GENDER"] not in GENDER:
            continue

        dev_mountain.append(row)
    return dev_mountain, dev_club
```